test/test_invalid_headers.py

```diff
@@ -365,6 +367,27 @@ def test_push_promise_skip_normalization(self, frame_factory, headers):
         )
         assert c.data_to_send() == pp_frame.serialize()
 
+    @pytest.mark.parametrize('headers', strippable_header_blocks)
+    def test_strippable_headers(self, frame_factory, headers):
+        """
+        Test connection related headers are removed before sending.
+        """
+        c = h2.connection.H2Connection()
+        c.initiate_connection()
+
+        # Clear the data, then try to send headers.
+        c.clear_outbound_data_buffer()
+        c.send_headers(1, headers)
+
+        conn_headers = (
+            'connection', 'proxy-connection', 'keep-alive',
+            'transfer-encoding', 'upgrade'
+        )
+        stripped_headers = [h for h in headers if h[0] not in conn_headers]
+
+        f = frame_factory.build_headers_frame(stripped_headers)
+        assert c.data_to_send() == f.serialize()
+
```

Comment:
I've combined `strippable_header_blocks` and `invalid_header_blocks` into an `all_header_blocks` variable which I'm now using for these tests. Since this normalization only runs in `normalize_outbound_headers` are we concerned about the `validate_inbound_headers` flag? `TestInvalidFrameSequences` should still cover the general ProxyErrors raised by receipt of these headers.